content/posts/eks-vs-ecs.md

```diff
@@ -0,0 +1,107 @@
+
+In December 2019, AWS started to [support using EKS with Fargate](https://aws.amazon.com/about-aws/whats-new/2019/12/run-serverless-kubernetes-pods-using-amazon-eks-and-aws-fargate/), so that specific differentiator between using ECS and EKS has vanished. Now you can deploy services in Kubernetes without worrying about the underlying infrastructure at all by using the Fargate serverless compute engine with EKS.
+
+Using EKS is in many ways similar to using ECS since whether you use Fargate or EC2, AWS will take care of provisioning the resources for you, letting you instead focus on the applications you will be running. The real differences come down to which container orchestrator you are using. ECS's solution or Kubernetes'.
+
+## Battle of the orchestrators
+
+If at the end of the day an application is going to be running in a container does it really matter which orchestrator you use as long as it works? Especially if one is simpler than the other? I would argue that it does matter. 
+
+Container orchestrators have become the new platform on which you build and deploy applications. While we often talk about vendor lock-in as a situation many people get into, it happens just as frequently with technologies. For instance, if you are developing for Linux machines your application isn't necessarily portable to a Windows machine. At least not without extra work. This is a form of lock-in.
+
+Investing time and effort into a container platform is locking you in to that platform to some degree. Sure your applications will be portable as they are containers, but the process and lifecycle will be different depending on which platform you use. This isn't to say that the migration between the two is complicated, just that it does require some time and effort.
+
+The focus of this post is about how this relates to CloudBees Core & Jenkins, but first let's do a quick pros/cons of both ECS and EKS. 
+
+### ECS Pros & Cons
+
+| Pros |
+| --- |
+| + Easy to use |
+| + Longstanding integration with Fargate to abstract away the infrastructure |
+| + Tight integration with the AWS ecosystem |
+
+| Cons |
+| --- |
+| - Proprietary tooling |
+| - Limited third-party ecosystem |
+| - Not portable outside of AWS |
+
+
+### EKS Pros & Cons
+
+| Pros |
+| --- |
+| + All the power of Kubernetes |
+| + Any Kubernetes work done is portable to any other Kubernetes cluster |
+| + Easy to build tooling around |
+| + New integration with Fargate to add serverless functionality* |
+
+
+| Cons |
+| --- |
+| - Kubernetes can be challenging to learn |
+| - Without automation or tooling, the process of deploying a single application is more involved |
+| - Can be overkill for small projects |
+
+
+* It's important to note that with the Fargate integration, there are some limitations such as which regions you can deploy to [https://docs.aws.amazon.com/eks/latest/userguide/fargate.html](https://docs.aws.amazon.com/eks/latest/userguide/fargate.html).
```

Comment:
Also, I thought that Fargate on EKS doesn't support stateful applications yet - so that is a drawback for specific types of applications - like Jenkins. Or has that changed?